scripts/benchmark_impl.py

```diff
@@ -0,0 +1,145 @@
+import os
+import datetime
+import sys
+import datetime
+import git
+from select import select
+
+from git import repo
+
+
+def __fill(exec, shared_para, data_type, report_path):
+    new_para = shared_para + " -fill=1 -type={}".format(data_type)
+    report = report_path + "fill"
+    print("Fill {}".format(data_type))
+    cmd = "{0} {1} > {2}".format(exec, new_para, report)
+    os.system(cmd)
+
+
+def read_random(exec, shared_para, data_type, report_path):
+    new_para = shared_para + \
+        " -fill=0 -type={} -read_ratio=1 -key_distribution=random".format(
+            data_type)
+    report = report_path + "read_random"
+    print("Read random {}".format(data_type))
+    cmd = "{0} {1} > {2}".format(exec, new_para, report)
+    os.system(cmd)
+
+
+def insert_random(exec, shared_para, data_type, report_path):
+    new_para = shared_para + \
+        " -fill=0 -type={} -read_ratio=0 -key_distribution=random -existing_keys_ratio=0".format(
+            data_type)
+    report = report_path + "insert_random"
+    print("Insert random {}".format(data_type))
+    cmd = "{0} {1} > {2}".format(exec, new_para, report)
+    os.system(cmd)
+
+
+def batch_insert_random(exec, shared_para, data_type, report_path):
+    assert data_type == "string"
+    new_para = shared_para + \
+        " -fill=0 -type={} -read_ratio=0 -batch=100 -key_distribution=random -existing_keys_ratio=0".format(
+            data_type)
+    report = report_path + "batch_insert_random"
+    print("Batch insert random {}".format(data_type))
+    cmd = "{0} {1} > {2}".format(exec, new_para, report)
+    os.system(cmd)
+
+
+def update_random(exec, shared_para, data_type, report_path):
+    new_para = shared_para + \
+        " -fill=0 -type={} -read_ratio=0 -key_distribution=random".format(
+            data_type)
+    report = report_path + "update_random"
+    print("Update random {}".format(data_type))
+    cmd = "{0} {1} > {2}".format(exec, new_para, report)
+    os.system(cmd)
+
+
+def read_write_random(exec, shared_para, data_type, report_path):
+    new_para = shared_para + \
+        " -fill=0 -type={} -read_ratio=0.9 -key_distribution=random".format(
+            data_type)
+    report = report_path + "read_write_random"
+    print("Read write random {}".format(data_type))
+    cmd = "{0} {1} > {2}".format(exec, new_para, report)
+    os.system(cmd)
+
+
+def range_scan(exec, shared_para, data_type, report_path):
+    new_para = shared_para + \
+        " -fill=0 -type={} -read_ratio=1 -scan=1".format(data_type)
+    report = report_path + "range_scan"
+    print("Range scan {}".format(data_type))
+    cmd = "{0} {1} > {2}".format(exec, new_para, report)
+    os.system(cmd)
+
+
+def confirm(dir):
+    timeout = 5
```

Comment:
5 seconds is too short.
Just press y+enter if you don't want to wait for a minute.

Reply:
done